psi4/src/psi4/dfocc/tensors.h

```diff
@@ -209,10 +209,24 @@ class Tensor2d {
     void pcopy(const SharedTensor2d &A, int dim_copy, int dim_skip);
     void pcopy(const SharedTensor2d &A, int dim_copy, int dim_skip, int start);
     double get_max_element();
-    // diagonalize: diagonalize via rsp
-    void diagonalize(const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff);
-    void diagonalize(const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff, bool ascending);
-    void diagonalize(int dim, const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff, bool ascending);
+
+    /// @brief diagonalizes the Tensor2d using LAPACK DSYEV
+    /// @param eigvectors matrix of eigenvectors (2D row major array, one column for each eigvector)
+    /// @param eigvalues array to hold eigenvalues
+    /// @param cutoff defunct argument, setting it is a no-op, will be removed in the future
+    /// @param ascending (optional, defaults true) Should the eigvalues be returned in ascending order?
+    void diagonalize(const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff,
+                     bool ascending = true);
+
+    /// @brief diagonalizes the Tensor2d using LAPACK DSYEV
+    /// @param dim number of rows (and columns)
+    /// @param eigvectors matrix of eigenvectors (2D row major array, one column for each eigvector)
+    /// @param eigvalues array to hold eigenvalues
+    /// @param cutoff defunct argument, setting it is a no-op, will be removed in the future
+    /// @param ascending Should the eigvalues be returned in ascending order?
+    void diagonalize(int dim, const SharedTensor2d &eigvectors, const SharedTensor1d &eigvalues, double cutoff,
+                     bool ascending);
```

Comment:
Would it make sense to have `ascending` for this overload default to `true`, as well?

Reply:
Probably yes, but I wanted to keep the interfaces as untouched as reasonable to minimize dfocc disruption for the time being. They will need to be changed anyways after the rest of dfocc gets merged.

Reply:
That's reasonable!